PWGJE/Tasks/emccellmonitor.cxx

```diff
@@ -0,0 +1,201 @@
+      case 15: supermoduleHistHelper<15>(col, row, amplitude, celltime); break;
+      case 16: supermoduleHistHelper<16>(col, row, amplitude, celltime); break;
+      case 17: supermoduleHistHelper<17>(col, row, amplitude, celltime); break;
+      case 18: supermoduleHistHelper<18>(col, row, amplitude, celltime); break;
+      case 19: supermoduleHistHelper<19>(col, row, amplitude, celltime); break;
+      default: break;
+    };
+  }
+
+  template<int supermoduleID>
+  void supermoduleHistHelper(int col, int row, double amplitude, double celltime) {
+    static constexpr std::string_view cellAmpHistSM[20] = {"cellAmplitudeSM/cellAmpSM0", "cellAmplitudeSM/cellAmpSM1", "cellAmplitudeSM/cellAmpSM2", "cellAmplitudeSM/cellAmpSM3", "cellAmplitudeSM/cellAmpSM4", "cellAmplitudeSM/cellAmpSM5", "cellAmplitudeSM/cellAmpSM6", "cellAmplitudeSM/cellAmpSM7", "cellAmplitudeSM/cellAmpSM8", "cellAmplitudeSM/cellAmpSM9", "cellAmplitudeSM/cellAmpSM10", "cellAmplitudeSM/cellAmpSM11", "cellAmplitudeSM/cellAmpSM12", "cellAmplitudeSM/cellAmpSM13", "cellAmplitudeSM/cellAmpSM14", "cellAmplitudeSM/cellAmpSM15", "cellAmplitudeSM/cellAmpSM16", "cellAmplitudeSM/cellAmpSM17", "cellAmplitudeSM/cellAmpSM18", "cellAmplitudeSM/cellAmpSM19"};
+    static constexpr std::string_view cellCountHistSM[20] = {"cellCountSM/cellAmpSM0", "cellCountSM/cellAmpSM1", "cellCountSM/cellAmpSM2", "cellCountSM/cellAmpSM3", "cellCountSM/cellAmpSM4", "cellCountSM/cellAmpSM5", "cellCountSM/cellAmpSM6", "cellCountSM/cellAmpSM7", "cellCountSM/cellAmpSM8", "cellCountSM/cellAmpSM9", "cellCountSM/cellAmpSM10", "cellCountSM/cellAmpSM11", "cellCountSM/cellAmpSM12", "cellCountSM/cellAmpSM13", "cellCountSM/cellAmpSM14", "cellCountSM/cellAmpSM15", "cellCountSM/cellAmpSM16", "cellCountSM/cellAmpSM17", "cellCountSM/cellAmpSM18", "cellCountSM/cellAmpSM19"};
+    static constexpr std::string_view cellAmpTimeHist[20] = {"cellAmplitudeTime/cellAmpTimeCorrSM0", "cellAmplitudeTime/cellAmpTimeCorrSM1", "cellAmplitudeTime/cellAmpTimeCorrSM2", "cellAmplitudeTime/cellAmpTimeCorrSM3", "cellAmplitudeTime/cellAmpTimeCorrSM4", "cellAmplitudeTime/cellAmpTimeCorrSM5", "cellAmplitudeTime/cellAmpTimeCorrSM6", "cellAmplitudeTime/cellAmpTimeCorrSM7", "cellAmplitudeTime/cellAmpTimeCorrSM8", "cellAmplitudeTime/cellAmpTimeCorrSM9", "cellAmplitudeTime/cellAmpTimeCorrSM10", "cellAmplitudeTime/cellAmpTimeCorrSM11", "cellAmplitudeTime/cellAmpTimeCorrSM12", "cellAmplitudeTime/cellAmpTimeCorrSM13", "cellAmplitudeTime/cellAmpTimeCorrSM14", "cellAmplitudeTime/cellAmpTimeCorrSM15", "cellAmplitudeTime/cellAmpTimeCorrSM16", "cellAmplitudeTime/cellAmpTimeCorrSM17", "cellAmplitudeTime/cellAmpTimeCorrSM18", "cellAmplitudeTime/cellAmpTimeCorrSM19"};
+    mHistManager.fill(HIST(cellAmpHistSM[supermoduleID]), col, row);
+    mHistManager.fill(HIST(cellCountHistSM[supermoduleID]), col, row, amplitude);
+    mHistManager.fill(HIST(cellAmpTimeHist[supermoduleID]), celltime, amplitude); 
+  }
```

Comment:
I have changed to the histogram registry but I am not fully happy with this part. The goal is that we have 3 histograms we want to monitor per supermodule ID. In this case the supermodule ID is determined dynamically from the tower ID and during the fill process becomes a dynamic part of the name. Based on my understanding of the histogram registry names must be constant. The only way around  I found so far is to have a templated helper function where the supermodule is a template parameter (and therefore constant) and internally 3 arrays with constant histogram names / supermodule. The mapping supermodule ID -> template parameter must therefore happen in a switch statement. This construction is a bit clumsy and inflexible, will still look for something better, maybe for the beginning it could go.

Reply:
Can the supermodule ID be used as an axis?

Reply:
These are already 2D histograms, going to higher dimensions would be possible but not really preferred. I can also imagine at some point other detectors might have similar interests (i.e. monitor stuff per tracking sector).